### DISCUSSION TITLE
Status Code 403/404 ?

### DISCUSSION THREAD

**Discussion by seanb3214** [jef/streetmerchant#1241] [Help] [answered]
## Description
What is status code 403 / 404? 

Thanks

**Comment by gigi2006** [comment]
IP Ban, maybe temp maybe permanent 
403

**Comment by seanb3214** [comment]
You're right, I ran it on a DaaS on AWS with no errors.

**Comment by jgramling17** [comment]
Just got a temporary ip ban from newegg for hitting their realTimeUpdates endpoint too often. Any workarounds or rate limits that wont trigger this?

**Comment by Utherellus** [comment]
I also would like to know that. Got banned on two IPs while other people how I know dont have this issue.

**Comment by DerHOgi** [comment]
The only way to avoid IP banns is to change the user agent string for each request an change the ip via proxy or vpn every few minutes.

**Reply by Utherellus** [reply]
any ideas how to automatize ip changer?

**Comment by DerHOgi** [comment]
PS: Amazon has a new nasty trick to take action against bots. If the server detects a lot of reloads within a certain period of time, products are shown as unavailable even if they are on stock. LOL

**Comment by greguren** [comment]
I am getting 404s and 403s on caseking but can access the store anyway on my regular browser. It doesn't seem like an IP ban, imho. At least with that store.

**Comment by jef** [comment]
> I am getting 404s and 403s on caseking but can access the store anyway on my regular browser. It doesn't seem like an IP ban, imho. At least with that store.

Run in headful mode to see whats going on: `HEADLESS="false"`.

**Comment by nazmi** [comment]
weird, run headless false. loaded cclonline.com perfectly when typing manually, fails on bot attempt in chromium. maybe needed some timeout to load cloudflare

**Comment by jef** [comment]
> weird, run headless false. loaded cclonline.com perfectly when typing manually, fails on bot attempt in chromium. maybe needed some timeout to load cloudflare

Fill out `USER_AGENT=""` with https://www.google.com/search?q=what+is+my+user+agent

https://jef.codes/streetmerchant/help/troubleshoot/#captcha-issues

That captcha issue can be also related to 403/404s.

**Comment by nazmi** [comment]
> Fill out `USER_AGENT=""` with https://www.google.com/search?q=what+is+my+user+agent
> 
> https://jef.codes/streetmerchant/help/troubleshoot/#captcha-issues
> 
> That captcha issue can be also related to 403/404s.

I did use my user agent. Edit: realised it's not 404 error, but you get the idea im not banned
![test](https://user-images.githubusercontent.com/524301/101534711-77538180-398f-11eb-82e4-7209090adde2.gif)

**Reply by 85256638** [reply]
@nazmi where to fill out the user-agent please?

**Reply by jef** [reply]
You shouldn't have to fill out a user-agent any longer.

**Comment by jef** [comment, marked as answer]
Good catch! I'll make some updates.

**Reply by jef** [reply]
It's in the backlog now. Thank you!

https://github.com/jef/streetmerchant/issues/1297
